### PR TITLE
Add Scheme-style OK tests

### PR DESCRIPTION
**Pull request by albert12132**
Adds support for Scheme-style tests. Specifically, this is meant to interface with UC Berkeley's [Scheme project](http://inst.eecs.berkeley.edu/~cs61a/fa14/proj/scheme/).

The change aren't quite done yet, but you can test out the current implementation by doing the following:

1. Copy the `scheme/` directory from the Scheme project (including the directory itself) into `<repo>/demo/scheme/`
2. Add a `__init__.py` file to the copied `scheme/` directory that contains the line `from scheme.scheme import *`
3. In all of the Scheme project files, add `scheme.` as a prefix to all imports related to the project (e.g. `scheme_reader` -> `scheme.scheme_reader`).
4. In `<repo>/demo/scheme`, run `ok-test`.

Please do NOT commit the copied scheme directory, as it may contain project solutions.

TODOs:

* [x] Implement `SchemeConsole.interact()`
* [ ] Add tests
* [ ] Add documentation

### CODE REVIEW

**Comment by albert12132** [comment]
This is ready to merge.

File: client/sources/ok_test/scheme.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+"""Console for interpreting Scheme. In particular, this is meant to integrate
+with UC Berkeley CS 61A's Scheme project. SchemeConsole expects the an
```
Comment (papajohn) [inline client/sources/ok_test/scheme.py]: expects [the] an importable...

File: client/sources/ok_test/scheme.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+importable module called "scheme" with the following interface:
+
+    scheme.create_global_frame()
+    scheme.read_eval_print_loop(next_line_fn, frame, interactive)
+    scheme.buffer_input()
+    scheme.read_line(code)
+    scheme.scheme_eval(exp, env)
```
Comment (Some3141) [inline client/sources/ok_test/scheme.py]: We should toss in a link to the scheme project so that it's obvious what these should do.

**Comment by papajohn** [comment]
Looks great!

**Pull request merged**